modules/appeals_api/app/controllers/concerns/appeals_api/pdf_downloads.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require 'common/file_helpers'
+require 'pdf_fill/filler'
+
+# rubocop:disable Metrics/ModuleLength
+module AppealsApi
+  module PdfDownloads
+    extend ActiveSupport::Concern
+
+    included do
+      def render_appeal_pdf_download(appeal, filename)
+        if expired?(appeal)
+          # If the veteran_icn is available, we can check it against the current request's ICN header to return a 404
+          # if it does not match, but otherwise, without auth_headers, we can't tell whether the current request is
+          # authorized, so in that case we return a specific error message regardless of authorization status.
+          if appeal.veteran_icn && appeal.veteran_icn != request.headers['X-VA-ICN']
+            raise ActiveRecord::RecordNotFound
+          else
+            return render_pdf_download_expired(appeal)
+          end
+        end
+
+        # We choose to return a 404 when PII in headers doesn't match the Appeal's PII so that we don't reveal the
+        # existence of records that the user can't access
+        raise ActiveRecord::RecordNotFound unless download_authorized?(appeal)
```

Comment:
I checked with cassowary about which status to return when the request headers don't match the appeal record [here](https://lighthouseva.slack.com/archives/C02SHTY8VNW/p1688076321644939), and it sounds like 404 is the way to go.